### ISSUE TITLE
RestfulFloodException should extend RestfulUnauthorizedException

### ISSUE THREAD

**Issue by ghost** [RESTful-Drupal/restful#719]
Hi,

We use HTTP basic auth on our development environments to keep Google out, but not for RESTful auth. The RestfulAuthenticationBasic class is returning a true for applies and authenticate keeps failing and therefore eventually it's overflowing flood.

I guess firstly, shouldn't we be able to turn off basic auth? We don't need it.

But secondly and more importantly, when I try to modify users now I'm getting a RestfulFloodException because the restful module tries to clear the entity cache and that asks for a user and that overflows the flood control. I think therefore that RestfulFloodException should extend RestfulUnauthorizedException which would at least fix my user save issues.

**Comment by amitaibu** [comment]
Related to #336 ?

**Comment by ghost** [comment]
Yes definitely sounds like it. Although I still think that RestfulFloodException should extend RestfulUnauthorizedException :smile:
